### DISCUSSION TITLE
default_classes save and restore?

### DISCUSSION THREAD

**Discussion by kfalcami** [zauberzeug/nicegui#3877] [Q&A] [answered]
### Question

One part of my UI has a bunch of labels and I want them all to have the same base classes(), which some of them will customize.  Other parts of my UI will have other base classes for labels.

I was going to use ui.label.default_classes('...') to set the base then restore it after the labels are made, but I don't see a way to access the existing default_classes so that I can restore them (except _default_classes which is protected).

I can see how I could sub-class ui.label adding get_default_classes(), or manage my own list/set of base classes, but I'm wondering if my use-case reveals an oversight in default_classes() or is just outside it's intended function/scope?

**Comment by falkoschindler** [comment, marked as answer]
Hi @kfalcami,

I think there's no way around subclassing (or helper functions) for creating custom elements. Default classes can be handy in certain situations, but don't cover all use cases.
